options.c

```diff
@@ -1169,6 +1195,49 @@ unmarshal_acl_list(region_type* r, struct buffer* b)
 	return list;
 }
 
+static void
+marshal_strv(struct buffer* b, char **strv)
+{
+	uint32_t i, n;
+
+	assert(b != NULL);
+
+	if (strv == NULL) {
+		marshal_u32(b, 0);
+		return;
+	}
+	for(n = 0; strv[n]; n++) {
+		/* do nothing */
+	}
+	marshal_u32(b, n);
+	for(i = 0; strv[i] != NULL; i++) {
+		marshal_str(b, strv[i]);
+	}
+	marshal_u8(b, 0);
```

Comment:
This marshal_u8 does not match with the unmarshal_strv routine.  That routine reads u32 and then a list of strings.  So that is what this routine should write, but it adds an unused zero byte.

Reply:
Sorry now I see that this adds the NULL at end of the array.